dataset/put-metadata.go

```diff
@@ -97,3 +98,77 @@ func putMetadata(w http.ResponseWriter, req *http.Request, dc DatasetClient, zc
 
 	log.Info(ctx, "put metadata: request successful", log.Data(logInfo))
 }
+
+// PutEditableMetadata updates a given list of metadata fields, agreed as being editable for both a dataset and a version object
+// This new endpoint makes an unique call to the dataset api updating only the relevant metadata fields in a transactional way
+// It also calls zebedee to update the collection
+func PutEditableMetadata(dc DatasetClient, zc ZebedeeClient) http.HandlerFunc {
+	return dphandlers.ControllerHandler(func(w http.ResponseWriter, r *http.Request, lang, collectionID, accessToken string) {
+		putEditableMetadata(w, r, dc, zc, accessToken, collectionID, lang)
+	})
+}
+
+func putEditableMetadata(w http.ResponseWriter, req *http.Request, dc DatasetClient, zc ZebedeeClient, userAccessToken, collectionID, lang string) {
```

Comment:
The lang parameter is not used: 
```suggestion
func putEditableMetadata(w http.ResponseWriter, req *http.Request, dc DatasetClient, zc ZebedeeClient, userAccessToken, collectionID) {
```